From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Timo Gurr <timo.gurr@gmail.com>
Date: Wed, 17 Jun 2020 16:41:55 +0200
Subject: [PATCH] spirv/glslang: fix compatibility with glslang >= 8.13.3743

---
 src/glsl/glslang.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/glsl/glslang.cc b/src/glsl/glslang.cc
index 5546120..fcd1745 100644
--- a/src/glsl/glslang.cc
+++ b/src/glsl/glslang.cc
@@ -211,6 +211,9 @@ const TBuiltInResource DefaultTBuiltInResource = {
     /* .maxTaskWorkGroupSizeZ_NV = */ 1,
     /* .maxMeshViewCountNV = */ 4,
 #endif
+#if GLSLANG_PATCH_LEVEL >= 3743
+    /* .maxDualSourceDrawBuffersEXT = */ 1,
+#endif
 
     /* .limits = */ {
         /* .nonInductiveForLoops = */ 1,
